text/1026-ember-data-deprecate-store-extends-ember-object.md

```diff
@@ -0,0 +1,109 @@
+---
+stage: accepted
+start-date: # In format 2024-05-11T00:00:00.000Z
+release-date: # In format YYYY-MM-DDT00:00:00.000Z
+release-versions:
+teams: # delete teams that aren't relevant
+  - data
+prs:
+  accepted: https://github.com/emberjs/rfcs/pull/1026
+project-link:
+suite: 
+---
+
+# EmberData | Deprecate Store extending EmberObject
+
+## Summary
+
+This RFC deprecates the Store extending from EmberObject. All EmberObject specific
+APIs included.
+
+## Motivation
+
+There are two motivations:
+
+First, extending EmberObject is vestigial. The Store makes no use of any EmberObject API,
+not even for use with Ember's container or service injection.
+
+Second, in order to support any Ember version, support any non-Ember framework, and support
+EmberData running in non-browser environments we want to remove unnecessary coupling to Ember the framework.
+
+## Detailed design
+
+Instead of deprecating every EmberObject method, we will feature flag the store extending
+EmberObject at the module level. This ensures the deprecation only prints once, and that
+once resolved the store will no longer extend thereby making it feasible to utilize the
+benefits of not extending EmberObject immediately.
```

Comment:
For consistency.

```suggestion
Instead of deprecating every EmberObject method, we will feature flag the Store extending
EmberObject at the module level. This ensures the deprecation only prints once, and that
once resolved the Store will no longer extend thereby making it feasible to utilize the
benefits of not extending EmberObject immediately.
```